### PR TITLE
Dala 4402 update eu taxonomy activity picker

### PR DESCRIPTION
**Pull request by dennis-dfine**
# Pull Request \Update EU Taxonomy Activity Picker
`Please also check if newly created wikipage https://github.com/d-fine/DatalandInternal/wiki/EU-Taxonomy-Update is understandable.`

## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
  - [x] If this PR includes work on the frontend, at least one `@ts-nocheck` is removed. Additionally, there should not be any `@ts-nocheck` in files modified by this PR. If no `@ts-nocheck` are left: Celebrate :tada: :confetti_ball: type-safety and remove this entry. 
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [x] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [x] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [x] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on the clone server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team
  - [x] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [ ] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [ ] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [ ] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on dev server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team

### CODE REVIEW

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/enums/eutaxonomy/nonfinancials/Activity.kt [hunk 1]
```diff
@@ -3,117 +3,161 @@ package org.dataland.datalandbackend.model.enums.eutaxonomy.nonfinancials
...
 /**
- * An enum that holds the different activites related to EU taxonomy for non financial companies
-*/
+ * An enum that holds the different activities related to EU taxonomy for non financial companies
+ */
 @Schema(
     enumAsRef = true,
 )
 enum class Activity(val value: String) {
+    AcquisitionAndOwnershipOfBuildings("Acquisition and ownership of buildings"),
```
Comment (LaurinStrelow) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/enums/eutaxonomy/nonfinancials/Activity.kt:12]: Are there any tests ensuring the consistency between Activity.kt, ActivityTree.ts and EuTaxonomyNames.ts?

File: dataland-frontend/tests/component/components/general/ActivityFormField.cy.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ describe('Component test for ActivitiesFormField', () => {
     cy.get('ul.p-treenode-children')
       .should('be.visible')
       .find('li.p-treenode')
-      .should('have.length', 7)
+      .should('have.length', 10)
       .eq(1)
```
Comment (LaurinStrelow) [inline dataland-frontend/tests/component/components/general/ActivityFormField.cy.ts:17]: Why is this needed?

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/enums/eutaxonomy/nonfinancials/Activity.kt (hunk 1 above)
Comment (LaurinStrelow) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/enums/eutaxonomy/nonfinancials/Activity.kt:12]: Also: Have any activities been removed or renamed with this change?

Comment (dennis-dfine) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/enums/eutaxonomy/nonfinancials/Activity.kt:12]: There are no tests at the moment, the whole taxonomy stuff is rather hacky in DevTools Repo.

No activities have been removed or renamed.

File: dataland-frontend/tests/component/components/general/ActivityFormField.cy.ts (hunk 2 above)
Comment (dennis-dfine) [reply dataland-frontend/tests/component/components/general/ActivityFormField.cy.ts:17]: The test checks how many items there are clustered in this specific Activities form field (I think it was "Construction something"). Since there were new Activities added to this cluster, the length increased.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/enums/eutaxonomy/nonfinancials/Activity.kt [hunk 3]
```diff
@@ -3,117 +3,161 @@ package org.dataland.datalandbackend.model.enums.eutaxonomy.nonfinancials
...
-    AnaerobicDigestionOfBioWaste("Anaerobic digestion of bio-waste"),
     CompostingOfBioWaste("Composting of bio-waste"),
-    MaterialRecoveryFromNonHazardousWaste("Material recovery from non-hazardous waste"),
-    LandfillGasCaptureAndUtilisation("Landfill gas capture and utilisation"),
-    TransportOfCo2("Transport of CO2"),
-    UndergroundPermanentGeologicalStorageOfCo2("Underground permanent geological storage of CO2"),
-    PassengerInterurbanRailTransport("Passenger interurban rail transport"),
+    ComputerProgrammingConsultancyAndRelatedActivities("Computer programming, consultancy and related activities"),
+    ConservationForestry("Conservation forestry"),
+    ConservationIncludingRestorationOfHabitats1Ecosystems2AndSpecies("Conservation, including restoration, of habitats[1], ecosystems[2] and species"),
```
[the remark this answers is not in the document]
Comment (dennis-dfine) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/enums/eutaxonomy/nonfinancials/Activity.kt:29]: Die Strings kommen so aus der taxonomy und sind nur Beschreibungen, das ist okay. Die liegen auch bei uns einfach in einem Freitext-Feld. Für den Parameternamen musste ich das anpassen, da stand es auch drin. Aber zumindest den String würde ich auch nicht ändern, denn beim nächsten Taxonomy-Update ist sonst vllt unklar, ob das eigentlich mal das gleiche war oder nich.
